policy-controller/k8s/index/src/inbound/index.rs

```diff
@@ -1695,27 +1717,37 @@ impl PolicyIndex {
         authzs
     }
 
-    fn http_routes<'p>(
+    fn routes<'p, MatchType>(
```

Comment:
I think the behavior of this function should actually be different between HTTP and GRPC because of how default routes work.  i.e. if no HttpRoute or GrpcRoute resources exist then the default routes should be added to the http_routes.  Therefore, I'd recommend not making this generic and instead having explicit http_routes and grpc_routes functions.  Common code can be factored into helper functions to reduce duplication.

Reply:
This change likely propagates up the stack and makes it so many of the functions in the index cannot be generic over route type.  I think this is okay because the way we treat these resources is different.  Duplication can be reduced by factoring out common code.

Reply:
Reverted